src/gui/editorwidgets/qgsdatetimeeditwrapper.cpp

```diff
@@ -113,8 +113,16 @@ void QgsDateTimeEditWrapper::showIndeterminateState()
 
 void QgsDateTimeEditWrapper::dateTimeChanged( const QDateTime& dateTime )
 {
-  const QString fieldFormat = config( "field_format", QGSDATETIMEEDIT_DATEFORMAT ).toString();
-  emit valueChanged( dateTime.toString( fieldFormat ) );
+  const bool fieldIsoFormat = config( "field_iso_format" , false ).toBool();
+  const QString fieldFormat = config( "field_format" , QGSDATETIMEEDIT_DATEFORMAT ).toString();
```

Comment:
Should in here a check for `field().type() == QVariant::DateTime` be done and in this case a QDateTeim emitted (instead of a string)? Just like on line 133

Reply:
see #5989 and #5990